modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/UniqueTokenFilterTests.java

```diff
@@ -54,4 +53,23 @@ protected TokenStreamComponents createComponents(String fieldName) {
 
         assertThat(test.incrementToken(), equalTo(false));
     }
+
+    public void testPositionIncrement() throws IOException {
+        Analyzer analyzer = new Analyzer() {
+            @Override
+            protected TokenStreamComponents createComponents(String fieldName) {
+                Tokenizer tokenizer = new MockTokenizer(MockTokenizer.WHITESPACE, false);
+                TokenFilter filters = new EdgeNGramTokenFilter(tokenizer, 1, 2, false);
+                filters = new UniqueTokenFilter(filters);
+                return new TokenStreamComponents(tokenizer, filters);
+            }
+        };
+
+        assertAnalyzesTo(analyzer, "foo bar bro",
+                new String[]{"f", "fo", "b", "ba", "br"},
```

Comment:
Could we also include multiple skipped tokens in here?  e.g. test for "foo bar bro bar bro baz"